server_spec.rb

```diff
@@ -0,0 +1,234 @@
+require_relative 'server'
+require_relative 'spec_helper'
+
+describe Server do
+  let(:app) { described_class.new }
```

Comment:
+1 for using `described_class` instead of an explicit name - but even nicer is to use `object`, which Rspec naively instantiated if the `describe` block targets a class.